tensorflow-framework/src/test/java/org/tensorflow/framework/metrics/AccuracyTest.java

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.metrics;
+
+import org.junit.jupiter.api.Test;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.utils.TestSession;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.core.Variable;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TInt32;
+
+public class AccuracyTest {
+  private final TestSession.Mode tfMode = TestSession.Mode.GRAPH;
+
+  @Test
+  public void testCorrect() {
+    try (TestSession session = TestSession.createTestSession(tfMode)) {
+      Ops tf = session.getTF();
+      Accuracy<TFloat32> instance = new Accuracy<>(tf, 1001L, TFloat32.class);
+      session.run(instance.resetStates());
```

Comment:
This sets the variables to their initial state.  The variables are `total`, and possibly `count`, in `Reduce`. I did not create these with an initializer so `tf.init()` won't work, so you have to run `resetStates()` to initialize the variables before your first attempt to add to them, thus the call to` session.run(instance.resetStates());` The idea he is that the initializers may differ in sub-classes of `Reduce`.

I did it this way in case there were multiple metrics that get created at various times, as it seems `tf.init() `is akin to the big bang. What I don't know is does `tf.init()` clear any registered initializers or not. I guess I need to test this.

Reply:
OK, I just tested `tf.init()` and it does **NOT** clear the initializers after it is run. So, it is a big bang kind of thing.